### PR TITLE
Rename Build to Action_builder

### PR DESCRIPTION
**Pull request by jeremiedimino**
So that we can reuse `Build` for `Memo.Build`, which will become the main build monad. Also, `Action_buider` seems like a better name for this module.

### CODE REVIEW

**Review by rgrinberg — APPROVED** [review]
This name doesn't quite roll of the tongue, but it's better than Build I suppose.

File: src/dune_engine/build_system.ml [hunk 1]
```diff
@@ -1176,23 +1179,23 @@ end = struct
       | Sandbox_config _ ->
         Fiber.return ())
 
-  module Build_exec = Build.Make_exec (struct
+  module Build_exec = Action_builder.Make_exec (struct
```
Comment (snowleopard) [inline src/dune_engine/build_system.ml:1182]: Should `Build_exec` become `Action_builder_exec`?

File: src/dune_engine/action_builder.mli [hunk 2]
```diff
@@ -54,8 +54,8 @@ with type 'a build := 'a t
     information. *)
 val set_file_system_accessors : file_exists:(Path.t -> bool) -> unit
 
-(** Add a set of targets to a build description, turning a target-less [Build.t]
-    into [Build.With_targets.t]. *)
+(** Add a set of targets to a build description, turning a target-less
```
Comment (snowleopard) [inline src/dune_engine/action_builder.mli:57]: The comments in this module say "build description" when referring to `t` which may now be confusing. Perhaps, we should replace "build description" with "action builder".

**Review by snowleopard — APPROVED** [review]
Looks good, thanks! I left a couple of minor comments.

File: src/dune_engine/build_system.ml (hunk 1 above)
Comment (jeremiedimino) [reply src/dune_engine/build_system.ml]: Indeed, I renamed it

File: src/dune_engine/action_builder.mli (hunk 2 above)
Comment (jeremiedimino) [reply src/dune_engine/action_builder.mli]: Agreed, I updated all the comments mentioning "build description"

**Comment by jeremiedimino** [comment]
> This name doesn't quite roll of the tongue, but it's better than Build I suppose.

True. Other suggestions still welcome :) At least, it won't be as annoying to rename `Action_builder` than it was to rename `Build` (because of `Path.Build`).

**Pull request merged**